source/btrfs/superblock.d

```
@@ -212,6 +213,32 @@ bool loadSuperblock(ref Superblock superblock, shared const void[] buffer) // no
     return false;
 }
 
+bool loadSuperblock(ref Superblock superblock, ref shared Device device) // nothrow
+{
+    foreach (offset; SUPERBLOCK_OFFSETS)
+    {
+        if (offset + SUPERBLOCK_SIZE < device.size)
+        {
+            try
+            {
+                auto devicePtr = device.dataPtr(offset, SUPERBLOCK_SIZE);
+                superblock.load(devicePtr);
+
+                if (!superblock.matches())
+                {
+                    return false;
+                }
+                return superblock.isValid();
+            }
+            catch (Exception e)
+            {
+                return false;
+            }
```

Bad idea to swallow `Exception`